coda_v2_python_client/firebase_client_wrapper.py

```diff
@@ -367,3 +367,86 @@ def set_segment_messages_metrics(self, segment_id, messages_metrics):
         :type metrics_map: core_data_modules.data_models.metrics.MessagesMetrics
         """
         self.get_segment_messages_metrics_ref(segment_id).set(messages_metrics.to_firebase_map())
+
+    def compute_segment_coding_progress(self, segment_id, messages=None, force_recount=False):
+        """
+        Compute and return the progress metrics for a given dataset.
+
+        This method will initialise the counts in Firestore if they do not already exist.
+
+        :param segment_id: Id of a segment.
+        :type segment_id: str
+        :param messages: list of core_data_modules.data_models.message.Message, defaults to None
```

Comment:
Can you explain also what will happen if you do/don't provide messages? i.e. that it will compute based on the provided messages or fetch them

Reply:
updated